pylint/checkers/non_ascii_names.py

```diff
@@ -0,0 +1,237 @@
+# Copyright (c) 2021-2022 Carli Freudenberg <carli.freudenberg@energymeteo.de>
+
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+"""All alphanumeric unicode character are allowed in Python but could do
+to similarities in their look, they can be confused.
+
+See: https://www.python.org/dev/peps/pep-0672/#confusable-characters-in-identifiers
+
+The following checkers are intended to make users are aware of these issues.
+"""
+
+import re
+import sys
+from typing import Iterable, List, Optional, Union
+
+from astroid import nodes
+
+import pylint.checkers
+from pylint import interfaces
+from pylint.constants import HUMAN_READABLE_TYPES
+from pylint.lint import PyLinter
+
+if sys.version_info >= (3, 8):
+    from typing import Protocol
+else:
+    from typing_extensions import Protocol
+
+
+if sys.version_info[:2] >= (3, 7):
+    # pylint: disable-next=fixme
+    # TODO: Remove after 3.6 has been deprecated
+    Py37Str = str
+else:
+
+    class Py37Str(str):
+        # Allow Python 3.6 compatibility
+        def isascii(self: str) -> bool:
+            return all("\u0000" <= x <= "\u007F" for x in self)
+
+
+class _AsciiOnlyCheckedNode(Protocol):
+    _is_ascii_only: bool
+
+
+class NonAsciiNamesChecker(pylint.checkers.BaseChecker):
+    """A strict name checker only allowing ASCII
+
+    If your programming guideline defines that you are programming in English,
+    then there should be no need for non ASCII characters in Python Names.
+    Everybody else can simply not use this check.
+
+    Note: This check only checks Names, so it ignores the content of
+          docstrings and comments!
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+    priority = -1
+
+    msgs = {
+        "C0144": (
+            '%s name "%s" contains a non-ASCII character',
+            "non-ascii-name",
+            (
+                "Used when the name contains at least one non-ASCII unicode character. "
+                "See https://www.python.org/dev/peps/pep-0672/#confusable-characters-in-identifiers"
+                " for a background why this could be bad."
+            ),
+        ),
+    }
+
+    name = "NonASCII-Checker"
+
+    def __init__(self, linter):
+        super().__init__(linter)
+        self._non_ascii_rgx_compiled = re.compile("[^A-Za-z0-9_]")
+
+    def _raise_name_warning(
+        self,
+        node: nodes.NodeNG,
+        node_type: str,
+        name: str,
+    ) -> None:
+        type_label = HUMAN_READABLE_TYPES[node_type]
+        args = (type_label.capitalize(), name)
+
+        self.add_message(
+            "non-ascii-name", node=node, args=args, confidence=interfaces.HIGH
+        )
+        self.linter.stats.increase_bad_name(node_type, 1)
```

Comment:
In the original code this is not increased for every `non-ascii-name`. Are you intentionally changing this?

Reply:
I will move the checker to a new PR and look again there.

Reply:
Thank you! I know this might have been a lot of more work than you expected when started to work on this, but with the age and size of `pylint`'s codebase we're very wary of increasing technical debt because of insufficient review. I really want to thank for all the effort you're putting in to this! 😄

Reply:
Removed